test/createCycleMiddleware.test.js

```diff
@@ -0,0 +1,70 @@
+import { createCycleMiddleware } from 'redux-cycle-middleware';
+import configureMockStore from 'redux-mock-store'
+import expect from 'expect'
+import xs from 'xstream';
+
+function initMockStore(...opts) {
+    const cycleMiddleware = createCycleMiddleware(...opts);
+    const middlewares = [ cycleMiddleware ]
+    const mockStore = configureMockStore(middlewares)
+    return mockStore;
+}
+
+describe('Redux cycle middleware', () => {
+  it('dispatches a PING to see whether the middleware dispatches a PONG', () => {
+    function main(sources) {
+      const pong$ = sources.ACTION
+        .filter(action => action.type === 'PING')
+        .mapTo({ type: 'PONG' });
+
+      return {
+        ACTION: pong$
+      }
+    }
+
+    const mockStore = initMockStore(main, {})
+
+    const expectedActions = [
+      { type: 'PING' },
+      { type: 'PONG' }
+    ]
+    const store = mockStore({ todos: [] })
+
+    store.dispatch({ type: 'PING' })
+
+    expect(store.getActions()).toEqual(expectedActions)
+  })
+
+  it('dispatches a PING to see whether the middleware dispatches a PONG after 1 second', () => {
+    function main(sources) {
+      const pong$ = sources.ACTION
+        .filter(action => action.type === 'PING')
+        .map(a =>
+            xs.periodic(1000)
+                .take(1)
+                .mapTo({ type: 'PONG' })
+        )
+        .flatten();
+
+      return {
+        ACTION: pong$
+      }
+    }
+
+    const mockStore = initMockStore(main, {})
+
+    const expectedActions = [
+      { type: 'PING' },
+      { type: 'PONG' }
+    ]
+    const store = mockStore({ todos: [] })
+
+    store.dispatch({ type: 'PING' })
+
+    // XXX since it's async (xs.periodic), we don't have actions here yet :(
+    // perhaps @cyclejs/time will solve this?
+    console.log('foo', store.getActions())
```

Comment:
In this specific case, can't we have the expectation inside `setTimeout(..., 1000)`? And one before that makes sure there's no PONG initially.

Since there's only one test right now that involves time, I think rolling an ad-hoc solution like that would be just fine.